File: 2.Algorithm/99.ETC/소수 찾기/소수 찾기.py
"""import math

def solution(n):
    answer = n-1
    for i in range(2,n+1):
        for j in range(2,int(math.sqrt(i))+1):
            if i % j == 0:
                answer -= 1
                break
    return answer"""

def solution(n):
    num = set(range(2,n+1))

    for i in range(2,n+1):
        if i in num:
            num -= set(range(2*i,n+1,i))
    return len(num)

File: 2.Algorithm/99.ETC/1.Math/1.Prime Number/소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = n-1;
        for(int i = 2; i <= n; i++){
            for(int j = 2; j < (int)Math.sqrt(i)+1; j++){
                if (i % j == 0){
                    answer -= 1;
                    break;
                }
            }
        }
        return answer;
    }
}